### DISCUSSION TITLE
SSR vs SPR app

### DISCUSSION THREAD

**Discussion by ghost** [nuxt/nuxt#8905] [Questions (Nuxt 2)]
I was wondering why or when I should choose SSR (`ssr: true, target: 'server'`) over SPR (`ssr: true, target: 'static'`)?
Most of the hosting solutions provide static sites (SPR) for free, while putting an app on a NodeJS server will cost a monthly fee.
So why should I use a NodeJS server for my Front-End?
When does it make sense?

**Comment by lewebsimple** [comment]
SSR allows dynamic routes to be served independently from the statically generated files hosted on the CDN (SSG). So if you have lots of dynamic pages that change often, you might save some build time / resources by hosting a nodejs server to generate them on-the-fly instead of re-building / re-deploying on every change.

**Reply by ghost** [reply]
@lewebsimple Thanks! Why should I save build time / resources? Waiting a few minutes to see the changes applied is OK. If hosting provider don‘t charge anything, I also don’t see any reason from this perspective.
I‘m thinking of a site with one new post/page every week: 1 (re-)build per week …

**Comment by udithishara** [comment]
It purely depends on your requirement, if your requirement is "new post/page everyweek" and comfortable building like that you can use it and it'll use less recourses as well

But if you want something with a login system where lot of the data is passed through an API, SSR will work better for that

**Comment by SaavyGirl** [comment]
On Fri, Mar 5, 2021 at 11:31 PM Udith Ishara Madusanka <
notifications@github.com> wrote:

> It purely depends on your requirement, if your requirement is "new
> post/page everyweek" and comfortable building like that you can use it and
> it'll use less recourses as well
>
> But if you want something with a login system where lot of the data is
> passed through an API, SSR will work better for that
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nuxt/nuxt.js/discussions/8905#discussioncomment-435276>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQR7RCZOKSCLFX23NU7ZD6TTCG4Z3ANCNFSM4YGUFT7A>
> .
>
-- 
Erica K Vance

**Comment by SaavyGirl** [comment]
On Sat, Mar 6, 2021 at 1:50 AM Erica Vance <vanceerica2@gmail.com> wrote:

>
>
> On Fri, Mar 5, 2021 at 11:31 PM Udith Ishara Madusanka <
> notifications@github.com> wrote:
>
>> It purely depends on your requirement, if your requirement is "new
>> post/page everyweek" and comfortable building like that you can use it and
>> it'll use less recourses as well
>>
>> But if you want something with a login system where lot of the data is
>> passed through an API, SSR will work better for that
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/nuxt/nuxt.js/discussions/8905#discussioncomment-435276>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQR7RCZOKSCLFX23NU7ZD6TTCG4Z3ANCNFSM4YGUFT7A>
>> .
>>
> --
> Erica K Vance
>
-- 
Erica K Vance
